bigchaindb/util.py

```diff
@@ -251,16 +373,87 @@ def verify_signature(signed_transaction):
         bool: True if the signature is correct, False otherwise.
     """
 
-    data = signed_transaction.copy()
+    for fulfillment in signed_transaction['transaction']['fulfillments']:
+        fulfillment_message = get_fulfillment_message(signed_transaction, fulfillment)
+        # verify the fulfillment (for now lets assume there is only one owner)
+        try:
+            parsed_fulfillment = cc.Fulfillment.from_uri(fulfillment['fulfillment'])
+        except Exception:
+            return False
+        is_valid = parsed_fulfillment.validate(serialize(fulfillment_message))
+
+        # if not a `CREATE` transaction
+        if fulfillment['input']:
```

Comment:
The `verify_signature` method doesn't really care what type of transaction it is. It first verifies if the fulfillments are correct and then it checks if the fulfillments actually match the conditions in the input, if a input exists